src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,165 @@
+package duke;
+
+import java.io.IOException;
+
+import java.util.regex.Pattern;
+
+/**
+ * Parses the response taken by user from Ui.
+ */
+public class Parser {
+
+	private static Pattern CHECKSTRING = Pattern.compile("-?\\d+");
+
+	private boolean isDone;
+
+	private Storage storage;
+	private TaskList tasks;
+
+	enum Update {
+		MARK,
+		UNMARK
+	}
+
+	enum Type {
+		TODO,
+		DEADLINE,
+		EVENT
+	}
+
+	/**
+	 * Constructor of a Parser class.
+	 *
+	 * @param storage The text doocument used for storing and loading data.
+	 * @param tasks The arraylist used to store tasks.
+	 */
+	public Parser(Storage storage, TaskList tasks) {
+		this.storage = storage;
+		this.tasks = tasks;
+		this.isDone = false;
+	}
+
+	/**
+	 * To check if an integer is given after user input "mark", "unmark" or "delete".
+	 *
+	 * @param strNum The String after user input "mark", "unmark" or "delete".
+	 * @return True if it is an integer, false otherwise.
+	 */
+	private static boolean isInteger(String strNum) {
+		if (strNum == null) {
+			return false;
+		}
+		return CHECKSTRING.matcher(strNum).matches();
+	}
+
+	/**
+	 * Checks if Duke is done running
+	 *
+	 * @return True when "bye" is inputted, otherwise false
+	 */
+	public boolean isItDone() {
+		return this.isDone;
+	}
+
+	/**
+	 * Loads tasks from text document into arraylist.
+	 *
+	 * @throws DukeException If incorrect information is used to add into the arraylist.
+	 * @throws IOException If the relative path to the file in invalid.
+	 */
+	public void load() throws DukeException, IOException {
+		this.storage.load();
+		this.storage.addDukeToList(this.tasks);
+	}
+
+	/**
+	 * runs command after receiving response from the user.
+	 *
+	 * @param response The string that consists of the user command.
+	 * @throws DukeException If the response is invalid.
+	 * @throws IOException If the relative path to the text document is invalid.
+	 */
+	public void reply(String response) throws DukeException, IOException {
+		String[] parts = response.split(" ", 2);
+		/** the action */
+		String part1 = parts[0];
+		String part2;
+		switch (part1) {
+		case "bye":
+			System.out.println("-----------------------------------------------");
+			System.out.println("Bye. Hope to see you again soon!");
+			System.out.println("-----------------------------------------------");
```

Comment:
Perhaps these few lines can be extracted out into the Ui.java class for a better segregation of functionality across classes.